hummingbot/connector/connector/balancer/balancer_connector.py

```diff
@@ -345,6 +319,79 @@ def stop_tracking_order(self, order_id: str):
         if order_id in self._in_flight_orders:
             del self._in_flight_orders[order_id]
 
+    async def _update_order_status(self):
+        """
+        Calls REST API to get status update for each in-flight order.
+        """
+        last_tick = self._last_poll_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL
+        current_tick = self.current_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL
+
+        if current_tick > last_tick and len(self._in_flight_orders) > 0:
+            tracked_orders = list(self._in_flight_orders.values())
+
+            tasks = []
+            for tracked_order in tracked_orders:
+                order_id = await tracked_order.get_exchange_order_id()
+                tasks.append(self._api_request("post",
+                                               "eth/get-receipt",
+                                               {"txHash": order_id}))
+            self.logger().info(f"Polling for order status updates of {len(tasks)} orders.")
+            update_results = await safe_gather(*tasks, return_exceptions=True)
+            for update_result in update_results:
+                if isinstance(update_result, Exception):
+                    raise update_result
+                if "txHash" not in update_result:
+                    self.logger().info(f"_update_order_status txHash not in resp: {update_result}")
+                    continue
+                if update_result["confirmed"] is True:
+                    if update_result["receipt"]["status"] == 1:
+                        gas_used = update_result["receipt"]["gasUsed"]
+                        gas_price = get_gas_price()
+                        fee = Decimal(str(gas_used)) * Decimal(str(gas_price)) / Decimal(str(1e9))
```

Comment:
Shouldn't the gas_used be the total amount of gas used for the transaction, ie. gas_used is the fee?

Reply:
gas_used is the actual amount used which should be lower than our gas limit estimation (i.e. number of swaps + base gas). The `get_gas_price()` returns the estimated (fast, fastest, etc) gas price. The fee (Transaction Fee) is the final cost used to complete the transaction.